src/v2/Components/ArtworkFilter/ArtworkFilters/__tests__/SizeFilter2.rtl.jest.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { ReactElement } from "react"
+import {
+  render as originalRender,
+  RenderOptions,
+  screen,
+} from "@testing-library/react"
+import userEvent from "@testing-library/user-event"
+
+import {
+  ArtworkFilterContextProps,
+  ArtworkFilterContextProvider,
+  useArtworkFilterContext,
+} from "v2/Components/ArtworkFilter/ArtworkFilterContext"
+import { SizeFilter2 } from "../SizeFilter2"
+
+const render = (ui: ReactElement, options: RenderOptions = {}) =>
+  originalRender(ui, { wrapper: Wrapper, ...options })
+
+const Wrapper: React.FC = ({ children }) => {
+  return (
+    <ArtworkFilterContextProvider>
+      <ClearAllButton />
+      {children}
+      <ArtworkFilterContextInspector />
+    </ArtworkFilterContextProvider>
+  )
+}
+
+const ClearAllButton: React.FC = () => {
+  const artworkFilterContext = useArtworkFilterContext()
+
+  return (
+    <button onClick={() => artworkFilterContext.resetFilters()}>
+      Clear all
+    </button>
+  )
+}
+
+const ArtworkFilterContextInspector: React.FC = () => {
+  const artworkFilterContext = useArtworkFilterContext()
+
+  return (
+    <aside data-testid="artwork-filter-context-inspector">
+      {JSON.stringify(artworkFilterContext, null, 2)}
+    </aside>
+  )
+}
```

Comment:
For a UI focused testing framework, how _do_ you make assertions about invisible application state?

My solution was to serialize it right into the UI of the test harness. If you use RTL's `screen.debug()` you see this:

<img width="600" alt="debug" src="https://user-images.githubusercontent.com/140521/119670885-47961a00-be07-11eb-9f74-f00043bac9b9.png">

This allows the test to "see" the same state that the user indirectly would.

Reply:
@dzucconi - didn't you do something similar to this in a hook test at some point? Context around that? I recall that it was an unpleasant work around...

Reply:
Yeah, [I've been doing this exact thing](https://github.com/artsy/eigen/blob/1cf8cebc3e2ec232197e5f1c512c42cc7bc4d2f5/src/lib/Components/ArtworkFilter/Filters/__tests__/SizeOptionsNew-tests.tsx#L29-L63). Feels kinda wrong but works well 👍